plume-models/src/comments.rs

```diff
@@ -216,42 +191,35 @@ impl FromActivity<Note, Connection> for Comment {
                     content: SafeString::new(
                         &note
                             .object_props
-                            .content_string()
-                            .expect("Comment::from_activity: content deserialization error"),
+                            .content_string()?
                     ),
                     spoiler_text: note
                         .object_props
                         .summary_string()
                         .unwrap_or_default(),
                     ap_url: note.object_props.id_string().ok(),
-                    in_response_to_id: previous_comment.clone().map(|c| c.id),
-                    post_id: previous_comment.map(|c| c.post_id).unwrap_or_else(|| {
-                        Post::find_by_ap_url(conn, previous_url)
-                            .expect("Comment::from_activity: post error")
-                            .id
-                    }),
-                    author_id: User::from_url(conn, actor.as_ref())
-                        .expect("Comment::from_activity: author error")
-                        .id,
+                    in_response_to_id: previous_comment.iter().map(|c| c.id).next(),
+                    post_id: previous_comment.map(|c| c.post_id)
+                        .or_else(|_| Ok(Post::find_by_ap_url(conn, previous_url)?.id) as Result<i32>)?,
+                    author_id: User::from_url(conn, actor.as_ref())?.id,
                     sensitive: false, // "sensitive" is not a standard property, we need to think about how to support it with the activitypub crate
                     public_visibility
                 },
-            );
+            )?;
 
             // save mentions
             if let Some(serde_json::Value::Array(tags)) = note.object_props.tag.clone() {
                 for tag in tags {
                     serde_json::from_value::<link::Mention>(tag)
-                        .map(|m| {
-                            let author = &Post::get(conn, comm.post_id)
-                                .expect("Comment::from_activity: error")
-                                .get_authors(conn)[0];
+                        .map_err(Error::from)
```

Comment:
https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err i read the doc, but still don't understand what this does

Reply:
I you understand `map`, this is basically the same thing, but for `Err(_)` instead of `Ok(_)`. And passing `Error::from` is the equivalent to `|e| e.from()`